### ISSUE TITLE
Forms wouldn't work "Mautic is not defined"

### ISSUE THREAD

**Issue by rafalsk** [mautic/mautic#10408]
### Mautic Version

4.0.0

### PHP version

7.4.19

### What happened?

![image](https://user-images.githubusercontent.com/9338163/132095984-00a7f63f-40ac-4ce8-962a-c1059bf375e8.png)

When visiting a web-url one generated and proposed by the Forms-generator, the above errors can be seen.
No other errors reported in browser.

It's an iframe with contents fully generated by back-end so difficult to blame anything on our side.

### How can we reproduce this issue?

Step 1: and above.

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by rafalsk** [comment]
Same situation occurs when using the word-press plugin, with the exception  that there's no error about Mautic being undefined.

LIVE sample [here](https://gridnet.org/mautic/form/1)

**Comment by RCheesley** [comment]
Do you have the Mautic tracking implemented correctly on the site and your CORS settings correct in Mautic?

This issue was identified and addressed in the General Availability release of Mautic 4. Please ensure that you are on the latest stable version and not on the Release Candidate version? Also try clearing your cache and rebuilding the forms if you have updated?

Note to team: probably a duplicate of resolved issue: https://github.com/mautic/mautic/issues/9951

**Comment by mabumusa1** [comment]
<del>This bug is confirmed, it shows on forms including preference center. </del>

I checked more into the issue, I think this has to deal with GrapeJS plugin injecting its assets to preference center page, although they are not required there. which cause the issue to show in Javascript console

**Comment by EnableServices** [comment]
Spotted an issue with mautic-form.js today which may be related.

Now the random string is added to the url, this section of code in mautic-form.js breaks (line 48, column 262):

```
var sParts=sjs[i].getAttribute("src").split("?");if(sParts[1])MauticSDK.setConfig(MauticSDK.parseToObject(sParts[1]));MauticSDK.initialize(sParts[0]);break;}}
```

because it assumes a well-formed query string with an = sign in the parseToObject function.

Interestingly, in mautic-form-src.js (in the same directory), this has been handled (line 833):

```
var sParts = sjs[i].getAttribute("src").split("?");
if (sParts[1] && sParts[1].indexOf("=") !== -1) MauticSDK.setConfig(MauticSDK.parseToObject(sParts[1]));
```

The fix for me was to update the section in mautic-form.js to match mautic-form-src.js (add ` && sParts[1].indexOf("=") !== -1` to the if).

In short, it looks like mautic-form.js hasn't recently been re-compiled from mautic-form-src.js

**Comment by EnableServices** [comment]
> Probably this fix is related #10509 Can you try it? It require `php bin/console m:a:g`

Looking at the code, I don't think it'll fix the specific issue because it's still expecting an '=' sign on which to split.

The issue I uncovered is triggered when the url {mautic_instance}/media/js/mautic-form.js is called with an anti-caching random string:

{mautic_instance}/media/js/mautic-form.js?abdefrder34

for instance. Here there **is** a query string but it's not the expected key=value&key=value format, and as there is just a key the JSON created is not valid.

**Comment by EnableServices** [comment]
> Probably this fix is related #10509 Can you try it? It require `php bin/console m:a:g`

Actually just this `php bin/console m:a:g` fixes the issue I mentioned.

**Comment by kuzmany** [comment]
Probably this fix is related https://github.com/mautic/mautic/pull/10509
Can you try it? It require `php bin/console m:a:g`
